packages/full-stack-tests/src/hierarchy-builder/grouping/LabelGrouping.test.ts

```diff
@@ -0,0 +1,360 @@
+
+    before(async function () {
+      await initialize();
+      subjectClassName = Subject.classFullName.replace(":", ".");
+      physicalPartitionClassName = PhysicalPartition.classFullName.replace(":", ".");
+      selectClauseFactory = new NodeSelectClauseFactory();
+    });
+
+    after(async () => {
+      await terminate();
+    });
+
+    function createProvider(props: { imodel: IModelConnection; hierarchy: IHierarchyLevelDefinitionsFactory }) {
+      const { imodel, hierarchy } = props;
+      const schemas = new SchemaContext();
+      schemas.addLocater(new ECSchemaRpcLocater(imodel.getRpcProps()));
+      const metadataProvider = createMetadataProvider(schemas);
+      return new HierarchyProvider({
+        metadataProvider,
+        hierarchyDefinition: hierarchy,
+        queryExecutor: createECSqlQueryExecutor(imodel),
+      });
+    }
+
+    const basicHierarchy: IHierarchyLevelDefinitionsFactory = {
+      async defineHierarchyLevel(parentNode) {
+        if (!parentNode) {
+          return [
+            {
+              fullClassName: subjectClassName,
+              query: {
+                ecsql: `
+                    SELECT ${await selectClauseFactory.createSelectClause({
+                      ecClassId: { selector: `this.ECClassId` },
+                      ecInstanceId: { selector: `this.ECInstanceId` },
+                      nodeLabel: "root subject",
+                    })}
+                    FROM ${subjectClassName} AS this
+                    WHERE this.ECInstanceId = (${IModel.rootSubjectId})
+                  `,
+              },
+            },
+          ];
+        } else if (HierarchyNode.isInstancesNode(parentNode) && parentNode.label === "root subject") {
+          return [
+            {
+              fullClassName: subjectClassName,
+              query: {
+                ecsql: `
+                    SELECT ${await selectClauseFactory.createSelectClause({
+                      ecClassId: { selector: `this.ECClassId` },
+                      ecInstanceId: { selector: `this.ECInstanceId` },
+                      nodeLabel: { selector: `this.UserLabel` },
+                      groupByLabel: true,
+                    })}
+                    FROM ${subjectClassName} AS this
+                    WHERE this.Parent.Id = (${IModel.rootSubjectId})
+                  `,
+              },
+            },
+          ];
+        }
+        return [];
+      },
+    };
+
+    it("does not create groups of 1", async function () {
+      const { imodel, ...keys } = await buildIModel(this, async (builder) => {
+        const rootSubject = { className: subjectClassName, id: IModel.rootSubjectId };
+        const childSubject1 = insertSubject({ builder, codeValue: "test subject 1", parentId: rootSubject.id, userLabel: "test1" });
+        const childSubject2 = insertSubject({ builder, codeValue: "test subject 2", parentId: rootSubject.id, userLabel: "test2" });
```

Comment:
Yes, in some tests labels must be the same. But because "codeValue" is used to create "code", all "codeValue" properties have to be unique. If we use codeValue as label, then we won't be able to create the same labels.